### ISSUE TITLE
PlatformColor crashes on Android

### ISSUE THREAD

**Issue by rosskhanas** [wix/react-native-navigation#6618]
### Issue Description

The app crashes when PlatformColor is defined via `?attr/XXX`.

### Steps to Reproduce / Code Snippets / Screenshots

```
Navigation.setDefaultOptions({
  backButton: {
    color: PlatformColor("?attr/colorOnBackground"),
  },
}
```

---
### Environment
* React Native Navigation version: 7.0.0
* React Native version: 0.63.2
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Debug

**Comment by guyca** [comment]
@rosskhanas Have your tried with one of the latest snapshot versions? `7.0.0-snapshot.1225` perhaps. I think this issue is already resolved by @danilobuerger

**Comment by rosskhanas** [comment]
@guyca hm, it seems not fixed. Everything works well when I use `PlatformColor("@color/XXX")`, but not when `?attr/XXX`.

Here are more details about the crash:

```
com.facebook.react.bridge.JSApplicationCausedNativeException: ColorValue: None of the paths in the `resource_paths` array resolved to a color resource.
    at com.facebook.react.bridge.ColorPropConverter.getColor(ColorPropConverter.java:71)
    at com.reactnativenavigation.options.parsers.ColorParser.parse(ColorParser.java:25)
    at com.reactnativenavigation.options.TitleOptions.parse(TitleOptions.java:31)
    at com.reactnativenavigation.options.TopBarOptions.parse(TopBarOptions.java:33)
    at com.reactnativenavigation.options.Options.parse(Options.java:22)
    at com.reactnativenavigation.react.NavigationModule.parse(NavigationModule.java:194)
    at com.reactnativenavigation.react.NavigationModule.lambda$setDefaultOptions$1$NavigationModule(NavigationModule.java:104)
    at com.reactnativenavigation.react.-$$Lambda$NavigationModule$N2JlaPHmovMYN5T38Ah4GsF6njg.run
    at com.reactnativenavigation.react.NavigationModule.lambda$handle$13$NavigationModule(NavigationModule.java:200)
    at com.reactnativenavigation.react.-$$Lambda$NavigationModule$_KMLiCFir6_k6qF7P2pzVizFOwk.run
    at android.os.Handler.handleCallback(Handler.java:873)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:193)
    at android.app.ActivityThread.main(ActivityThread.java:6762)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by guyca** [comment]
@rosskhanas Thanks. What's the exact color value you're trying to use? What's `XXX` ?

**Comment by rosskhanas** [comment]
@guyca I tried several of them, for example `PlatformColor("?attr/colorOnBackground")` which is defined as `<item name="colorOnBackground">@color/on_background_light</item>` in `styles.xml`.

**Comment by danilobuerger** [comment]
@rosskhanas If its defined as an xml resource, using `@` is correct. The `?` is for style references to the current theme. Also see the implementation detail here: https://github.com/facebook/react-native/pull/27908

**Comment by guyca** [comment]
@danilobuerger Thanks for looking into this! 💯

**Comment by rosskhanas** [comment]
@danilobuerger right, `PlatformColor("?attr/colorOnBackground")` is the way to define theme-specific colors, correct? This way of using colors is supported in React Native, but once it is used for setting navigation colors, the app crashes...

**Comment by danilobuerger** [comment]
@rosskhanas Even so, `PlatformColor` is coming from react-native. The only time we can accept it as a bug here is if you can use the same `PlatformColor` on a react-native view element (for example `<Text style={color: PlatformColor("?attr/colorOnBackground")}>...</Text>`) without it crashing and it working and if you can provide a minimal reproduction showing that the same `PlatformColor` is working on a react-native view element but not working when passed through react-native-navigation.

**Comment by rosskhanas** [comment]
@danilobuerger exactly, this color works with `View` but not when passed through react-native-navigation. Here is a link to the repo: https://github.com/rosskhanas/platformcolor/tree/7e3a8fa0efd34ea07f8c4f50d588d741329625c6

Here is the line you can uncomment to get the app crashed: https://github.com/rosskhanas/platformcolor/blob/7e3a8fa0efd34ea07f8c4f50d588d741329625c6/index.js#L22

**Comment by danilobuerger** [comment]
Awesome, thanks for the reproduction

**Comment by guyca** [comment]
Thanks for the reproduction @rosskhanas, I'm on vacation for the next two weeks. Will get to this when I get back.

**Comment by rosskhanas** [comment]
@guyca did you have a chance to look at this? 🥺 Thank you.

**Comment by rosskhanas** [comment]
@guyca I wish I could, I'm not an Android expert 🙈 I'm open to providing some testing assistance if needed.

**Comment by habovh** [comment]
Any news on this? I'm having this issue on `7.18.1`.

**Comment by guyca** [comment]
@habovh @rosskhanas I'm not actively involved in the project anymore. I believe that if you're using `PlatformColor` to support dark mode then you should be using RNNColor as outlined [here](https://wix.github.io/react-native-navigation/next/docs/style-theme#dark-mode).

**Comment by habovh** [comment]
@guyca thanks for your reply! This would work fine for custom, app-defined colors, but this still prevents from using actual PlatformColors so the app blends in with other native apps.

**Comment by guyca** [comment]
@swabbass I believe this was overlooked in the design. How can users use the default system dark/light colors for background/text?

**Comment by swabbass** [comment]
@guyca  thx for mentioning me, I'll take it from here.

@habovh  The usage of `PlatformColor` was introduced in order to make apps use natively declared colours like in their res/*.xml on Android, now that is per app, in order for an app to blend with other apps, they need to use the same theme, for example, system default colours like `?android:attr/textColor` , which is common across system apps like Settings, or in a Hybrid app that uses native as the main views and some parts ReactNative.

So blending is not a good reason, but if an app wants to set the theme on the native side that is fine we are not preventing that usage, the easiest way is to export colours via Constants (NativeModule) from the native side and then use our colour solution because the processing of these type of colours is done in the [native side](https://github.com/facebook/react-native/blob/1465c8f3874cdee8c325ab4a4916fda0b3e43bdb/ReactAndroid/src/main/java/com/facebook/react/bridge/ColorPropConverter.java#L24) and we need to add support for that, which will take some time.

tl;dr
There are pros and cons to this way of providing theme aware colours, we discussed this internally before we added support for the dark theme, and our solution is to declare the colours once, in the JS, and use it natively, which is a dynamic way to define themes at least for our components, taking care of theme changes, without the need to recompile the apps to change a colour value, with less client code.

Here are my thoughts about `PlatformColor`:

Pros: (maybe there is more)
 - Apps (hybrid ones) that still have themes defined in native and want to use it in ReactNative.
 - No need to take care of system UI mode changes.
 
Cons:
- Duplicate definitions across hosts (Android/iOS..).
- Updating colours demand to recompile 2 binaries.
- No unified colour naming on the native side, which leads to duplicate JS code like [here](https://reactnative.dev/docs/platformcolor#example).
- Defies ReactNative main idea, to write once...

**Comment by guyca** [comment]
@swabbass Can RNNColor support "defaultTextColor" (?android:attr/textColor) , "defaultBackgroundColor" (?android:colorBackground
) and resolve them at runtime in native?

**Comment by swabbass** [comment]
> defaultBackgroundColor

definitely, I'm looking for a solution that our `Color` object takes the `PlatformColor` return type, and handle it natively, bypassing `optionsProcessor` calling [`ColorPropConverter`](https://github.com/facebook/react-native/blob/v0.64.2/ReactAndroid/src/main/java/com/facebook/react/bridge/ColorPropConverter.java#L24) in the native side.

**Comment by habovh** [comment]
@swabbass sounds like best of both worlds!

Thanks for your detailed message earlier, it definitely got me thinking about the way we handle colors. The main reason we're using PlatformColors is so that we can avoid re-rendering our JS components where the native can just update the displayed colors. To us, avoiding extra render cycles is outweighing the duplicated color definitions.

**Comment by danilobuerger** [comment]
Using something like https://github.com/klarna-incubator/platform-colors lets you define the colors in one place anyway. The rest is just generated code.

**Comment by swabbass** [comment]
@danilobuerger Looks cool, thx for the suggestion, it can be helpful, but I'm a little suspicious about generated code, it is a moving part.
